### PR TITLE
UT: Serialize all test CFGs

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs [hunk 1]
```diff
@@ -102,13 +103,16 @@ End Sub
                 {
                     throw new ArgumentException($"Anonymous function with '{anonymousFunctionFragment}' fragment was not found.");
                 }
-                return cfg.GetAnonymousFunctionControlFlowGraph(anonymousFunction, default);
-            }
-            else
-            {
-                return cfg;
+                cfg = cfg.GetAnonymousFunctionControlFlowGraph(anonymousFunction, default);
             }
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs:107]: Line 91:106 could be rewritten as (untested code)
```cs
            cfg = (localFunctionName, anonymousFunctionFragment) switch
            {
                (null, null) => cfg,
                ({ } localFunction, null) => cfg.GetLocalFunctionControlFlowGraph(cfg.LocalFunctions.Single(x => x.Name == localFunctionName), default),
                (null, { } fragment) =>
                    cfg.FlowAnonymousFunctionOperations().SingleOrDefault(x => x.WrappedOperation.Syntax.ToString().Contains(anonymousFunctionFragment)) is { WrappedOperation: { } } anonymousFunction
                        ? cfg.GetAnonymousFunctionControlFlowGraph(anonymousFunction, default)
                        : throw new ArgumentException($"Anonymous function with '{anonymousFunctionFragment}' fragment was not found."),
                _ => throw new InvalidOperationException($"Specify {nameof(localFunctionName)} or {nameof(anonymousFunctionFragment)}."),
            };
```

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. You may want to rewrite the if statements as suggested.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs:107]: That's hard to follow :/
